verify.js

```diff
@@ -153,7 +153,12 @@ module.exports = function (jwtString, secretOrPublicKey, options, callback) {
       }
     }
 
-    if (options.audience) {
+    if (typeof options.audience === 'function') {
+      var match = options.audience(payload.aud)
```

Comment:
Why set this variable at all? Wasted memory allocation when the processor is only using it once on the next line.